README.md
# Our Sem 1 Assignment- Make a simple search engine
### Credit to my teammates!
Thanks to Fahad, Yong Ming, Wei Hao and Srija for contributing to the project so we can have this wonderful search engine presented to our 
lecturer.
<br/>
The part I coded is the logic behing calculator and tic-tac-toc.


News.txt
1. Reddit.com
Reddit is a network of communities based on people's interests. Find communities you're interested in, and become part of an online community!
2. Nytimes.com
The New York Times - Breaking News, World News & Multimedia
The New York Times: Find breaking news, multimedia, reviews & opinion on Washington, business, sports, movies, travel, books, jobs, education, real estate, cars & more at nytimes.com.
3. News.google.com
Google News
Comprehensive, up-to-date news coverage, aggregated from sources all over the world by Google News.
4. Cnn.com
CNN - Breaking News, Latest News and Videos
View the latest news and breaking news today for U.S., world, weather, entertainment, politics and health at CNN.com.
5. Theguardian.com
News, sport and opinion from the Guardian's global edition | The Guardian
Latest international news, sport and comment from the Guardian
6. Shutterstock.com
Stock Images, Photos, Vectors, Video, and Music | Shutterstock
Search millions of royalty free stock images, photos, videos, and music. Get inspired by 1000s of new, high-resolution stock images added daily.
7. Indiatimes.com
Trending stories on Indian Lifestyle, Culture, Relationships, Food, Travel, Entertainment, News and New Technology News - Indiatimes.com
Indiatimes.com brings you the news, articles, stories and videos on entertainment, latest lifestyle, culture and new technologies emerging worldwide.
8. Washingtonpost.com
Washington Post: Breaking News, World, US, DC News & Analysis - The Washington Post
The Washington Post
Breaking news and analysis on politics, business, world national news, entertainment more. In-depth DC, Virginia, Maryland news coverage including traffic, weather, crime, education, restaurant reviews and more.
9. News.yahoo.com
Yahoo News - Latest News & Headlines
The latest news and headlines from Yahoo! News. Get breaking news stories and in-depth coverage with videos and photos.
10. Forbes.com
Forbes
Forbes is a global media company, focusing on business, investing, technology, entrepreneurship, leadership, and lifestyle.
11. Foxnews.com
Fox News - Breaking News Updates | Latest News Headlines | Photos & News Videos
Breaking News, Latest News and Current News from FOXNews.com. Breaking news and video. Latest Current News: U.S., World, Entertainment, Health, Business, Technology, Politics, Sports.
12. Weather.com
Ramalan dan Keadaan Cuaca di Kuala Lumpur - The Weather Channel | Weather.com
Ramalan cuaca hari ini dan malam ini di , keadaan cuaca dan radar Doppler daripada The Weather Channel dan Weather.com
13. Timesofindia.indiatimes.com
News: Breaking News, National news, Latest Bollywood News, Sports News, Business News and Political News | Times of India
News Today: Read Today's Live News on Sports, Business, Entertainment, Blogs and Opinions from leading columnists. Times of India brings the Breaking News and Latest News Headlines from India and around the World.
14. Economictimes.indiatimes.com
Business News Live, Share Market News - Read Latest Finance News, IPO, Mutual Funds News
Business News- Read Latest Financial news, Stock/Share Market News, Economy News, Business News on The Economic Times.  Find IPO Analysis, Mutual Funds Trends & Analysis, Gold Rate, Real Estate & more. 
15. Cnbc.com
International Business, World News & Global Stock Market Analysis
CNBC International is the world leader for news on business, technology, China, trade, oil prices, the Middle East and markets.
16. Bloomberg.com
Bloomberg - Are you a robot?
Bloomberg.com
17. Usatoday.com
USA TODAY: Latest World and US News  - USATODAY.com
USA TODAY delivers current local and national news, sports, entertainment, finance, technology, and more through award-winning journalism, photos, videos and VR.
18. Socket Timeout exception: Accuweather.comRead timed out
19. Chinadaily.com.cn
20. Reuters.com
Business & Financial News, U.S & International Breaking News  | Reuters
Reuters.com brings you the latest news from around the world, covering breaking news in markets, business, politics, entertainment, technology, video and pictures.
21. Wsj.com
The Wall Street Journal - Breaking News, Business, Financial & Economic News, World News and Video
Breaking news and analysis from the U.S. and around the world at WSJ.com. Politics, Economics, Markets, Life & Arts, and in-depth reporting.
22. Chron.com
Houston Local News, Weather, Headlines, Sports, Business, and Entertainment - The Houston Chronicle at Chron.com - Houston Chronicle
Breaking news headlines and Houston weather on Chron.com. Houston news, events, plus Astros and Rockets from the online home of the Houston Chronicle.
23. Nypost.com
Your source for breaking news, news about New York, sports, business, entertainment, opinion, real estate, culture, fashion, and more.
24. Nbcnews.com
25. Dw.com
News and current affairs from Germany and around the world | DW
News, off-beat stories and analysis of German and international affairs. Dive deeper with our features from Europe and beyond. Watch our 24/7 TV stream.
26. Usnews.com
27. Drudgereport.com
DRUDGE REPORT 2020®
28. News.com.au
news.com.au — Australia’s #1 news site
We say what people are thinking and cover the issues that get people talking balancing Australian and global moments — from politics to pop culture.
29. Wunderground.com
Local Weather Forecast, News and Conditions | Weather Underground
Weather Underground provides local & long-range weather forecasts, weather reports, maps & tropical weather conditions for locations worldwide
30. Cbsnews.com
Live, breaking news today: Latest national headlines, world news and more from CBSNews.com and watch the CBSN live news stream 24x7
Watch CBSN the live news stream from CBS News and get the latest, breaking news headlines of the day for national news and world news today.
31. Hindustantimes.com
News Headlines, English News, Today Headlines, Top Stories | Hindustan Times

src/tictactoe/translator2.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package tictactoe;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.logging.Level;
import java.util.logging.Logger;

/**
 *
 * @author fahad
 */
public class translator2 extends javax.swing.JFrame {

    private static final String CLIENT_ID = "FREE_TRIAL_ACCOUNT";
    private static final String CLIENT_SECRET = "PUBLIC_SECRET";
    private static final String ENDPOINT = "http://api.whatsmate.net/v1/translation/translate";
    
    public translator2() {
        initComponents();
    }

    /**
     * This method is called from within the constructor to initialize the form.
     * WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the Form Editor.
     */
    @SuppressWarnings("unchecked")
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {

        jLabel1 = new javax.swing.JLabel();
        jSeparator1 = new javax.swing.JSeparator();
        jComboBox1 = new javax.swing.JComboBox<>();
        jComboBox2 = new javax.swing.JComboBox<>();
        jLabel2 = new javax.swing.JLabel();
        jLabel3 = new javax.swing.JLabel();
        t1 = new javax.swing.JTextField();
        t2 = new javax.swing.JTextField();
        jButton1 = new javax.swing.JButton();
        jButton2 = new javax.swing.JButton();
        jLabel4 = new javax.swing.JLabel();

        setDefaultCloseOperation(javax.swing.WindowConstants.DISPOSE_ON_CLOSE);
        setTitle("Googol Translate");
        setResizable(false);
        getContentPane().setLayout(new org.netbeans.lib.awtextra.AbsoluteLayout());

        jLabel1.setFont(new java.awt.Font("Tw Cen MT", 0, 24)); // NOI18N
        jLabel1.setForeground(new java.awt.Color(255, 255, 255));
        jLabel1.setText("GOOGOL Translate");
        getContentPane().add(jLabel1, new org.netbeans.lib.awtextra.AbsoluteConstraints(209, 15, 216, -1));
        getContentPane().add(jSeparator1, new org.netbeans.lib.awtextra.AbsoluteConstraints(0, 52, 642, 10));

        jComboBox1.setFont(new java.awt.Font("Tw Cen MT", 0, 18)); // NOI18N
        jComboBox1.setModel(new javax.swing.DefaultComboBoxModel<>(new String[] { "English", "Malay", "Indonesian", "Spanish", "French" }));
        getContentPane().add(jComboBox1, new org.netbeans.lib.awtextra.AbsoluteConstraints(85, 114, -1, -1));

        jComboBox2.setFont(new java.awt.Font("Tw Cen MT", 0, 18)); // NOI18N
        jComboBox2.setModel(new javax.swing.DefaultComboBoxModel<>(new String[] { "English", "Malay", "Indonesian", "Spanish", "French" }));
        getContentPane().add(jComboBox2, new org.netbeans.lib.awtextra.AbsoluteConstraints(431, 114, -1, -1));

        jLabel2.setFont(new java.awt.Font("Tw Cen MT", 0, 14)); // NOI18N
        jLabel2.setForeground(new java.awt.Color(255, 255, 255));
        jLabel2.setText("Translate From :");
        getContentPane().add(jLabel2, new org.netbeans.lib.awtextra.AbsoluteConstraints(96, 80, 100, -1));

        jLabel3.setFont(new java.awt.Font("Tw Cen MT", 0, 14)); // NOI18N
        jLabel3.setForeground(new java.awt.Color(255, 255, 255));
        jLabel3.setText("Translate To :");
        getContentPane().add(jLabel3, new org.netbeans.lib.awtextra.AbsoluteConstraints(451, 80, 100, -1));

        t1.setFont(new java.awt.Font("Tw Cen MT", 0, 14)); // NOI18N
        getContentPane().add(t1, new org.netbeans.lib.awtextra.AbsoluteConstraints(26, 165, 251, -1));

        t2.setEditable(false);
        t2.setFont(new java.awt.Font("Tw Cen MT", 0, 14)); // NOI18N
        getContentPane().add(t2, new org.netbeans.lib.awtextra.AbsoluteConstraints(357, 165, 251, -1));

        jButton1.setFont(new java.awt.Font("Tw Cen MT", 0, 18)); // NOI18N
        jButton1.setText("Translate");
        jButton1.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                jButton1ActionPerformed(evt);
            }
        });
        getContentPane().add(jButton1, new org.netbeans.lib.awtextra.AbsoluteConstraints(176, 234, -1, -1));

        jButton2.setFont(new java.awt.Font("Tw Cen MT", 0, 18)); // NOI18N
        jButton2.setText("Reset");
        jButton2.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                jButton2ActionPerformed(evt);
            }
        });
        getContentPane().add(jButton2, new org.netbeans.lib.awtextra.AbsoluteConstraints(357, 234, -1, -1));

        jLabel4.setIcon(new javax.swing.ImageIcon(getClass().getResource("/tictactoe/263385-space_shuttle-minimalism-portrait_display-748x1197.jpg"))); // NOI18N
        getContentPane().add(jLabel4, new org.netbeans.lib.awtextra.AbsoluteConstraints(-50, -770, 700, 1080));

        pack();
    }// </editor-fold>//GEN-END:initComponents

    private void jButton1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jButton1ActionPerformed
        // TODO add your handling code here:
        String fromLang="";
        String toLang="";
        switch(jComboBox1.getSelectedIndex()){
            case 0:
                fromLang = "en";
                break;
            case 1: 
                fromLang = "ms";
                break;
            case 2:
                fromLang = "id";
                break;
            case 3: 
                fromLang = "es";
                break;
            case 4: 
                fromLang = "fr";
                break; 
           }
        switch(jComboBox2.getSelectedIndex()){
            case 0:
                toLang = "en";
                break;
            case 1: 
                toLang = "ms";
                break;
            case 2:
                toLang = "id";
                break;
            case 3: 
                toLang = "es";
                break;
            case 4: 
                toLang = "fr";
                break;      
        }
        String text = t1.getText();
        
        try {
            translate(fromLang, toLang, text);
        } catch (Exception ex) {
            Logger.getLogger(translator2.class.getName()).log(Level.SEVERE, null, ex);
        }
    }//GEN-LAST:event_jButton1ActionPerformed

    private void jButton2ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jButton2ActionPerformed
        // TODO add your handling code here:
        t1.setText("");
        t2.setText("");
        jComboBox1.setSelectedItem(-1);
        jComboBox2.setSelectedItem(-1);
        
    }//GEN-LAST:event_jButton2ActionPerformed

    public void translate(String fromLang, String toLang, String text) throws Exception{
        String jsonPayload = new StringBuilder()
      .append("{")
      .append("\"fromLang\":\"")
      .append(fromLang)
      .append("\",")
      .append("\"toLang\":\"")
      .append(toLang)
      .append("\",")
      .append("\"text\":\"")
      .append(text)
      .append("\"")
      .append("}")
      .toString();
        
        URL url = new URL(ENDPOINT);
        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
        conn.setDoOutput(true);
        conn.setRequestMethod("POST");
        conn.setRequestProperty("X-WM-CLIENT-ID", CLIENT_ID);
        conn.setRequestProperty("X-WM-CLIENT-SECRET", CLIENT_SECRET);
        conn.setRequestProperty("Content-Type", "application/json");
        
        OutputStream os = conn.getOutputStream();
        os.write(jsonPayload.getBytes());
        os.flush();
        os.close();
        
        int statusCode = conn.getResponseCode();
        
        BufferedReader br = new BufferedReader(new InputStreamReader(
        (statusCode == 200) ? conn.getInputStream() : conn.getErrorStream()
      ));
        
        String output;
        while ((output = br.readLine()) != null) {
            System.out.println(output);
            t2.setText(output);
    }
    conn.disconnect();
    
    }
    /**
     * @param args the command line arguments
     */
    public static void main(String args[]) {
        /* Set the Nimbus look and feel */
        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
        /* If Nimbus (introduced in Java SE 6) is not available, stay with the default look and feel.
         * For details see http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html 
         */
        try {
            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
                if ("Nimbus".equals(info.getName())) {
                    javax.swing.UIManager.setLookAndFeel(info.getClassName());
                    break;
                }
            }
        } catch (ClassNotFoundException ex) {
            java.util.logging.Logger.getLogger(translator2.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (InstantiationException ex) {
            java.util.logging.Logger.getLogger(translator2.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (IllegalAccessException ex) {
            java.util.logging.Logger.getLogger(translator2.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
            java.util.logging.Logger.getLogger(translator2.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        }
        //</editor-fold>

        /* Create and display the form */
        java.awt.EventQueue.invokeLater(new Runnable() {
            public void run() {
                new translator2().setVisible(true);
            }
        });
    }

    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JButton jButton1;
    private javax.swing.JButton jButton2;
    private javax.swing.JComboBox<String> jComboBox1;
    private javax.swing.JComboBox<String> jComboBox2;
    private javax.swing.JLabel jLabel1;
    private javax.swing.JLabel jLabel2;
    private javax.swing.JLabel jLabel3;
    private javax.swing.JLabel jLabel4;
    private javax.swing.JSeparator jSeparator1;
    private javax.swing.JTextField t1;
    private javax.swing.JTextField t2;
    // End of variables declaration//GEN-END:variables
}
